avahi-common/alternative.c

```diff
@@ -109,6 +113,13 @@ char *avahi_alternative_host_name(const char *s) {
         avahi_free(c);
     }
 
+    alt = alternative;
+    len = sizeof(alternative);
+    ret = avahi_escape_label(r, strlen(r), &alt, &len);
+
+    avahi_free(r);
+    r = avahi_strdup(ret);
```

Comment:
`r` can be `NULL` so it should be checked probably.

Reply:
Looks like the check isn't necessary here. NULL is either caught by the assertion or just returned if assertions are turned off.